app/controllers/events_controller.rb

```diff
@@ -14,10 +14,15 @@ class EventsController < ApplicationController
   # GET /events
   # GET /events.json
   def index
-    # Duration to view - default to day view
-    @period = params[:period].to_s || 'day'
+    # Duration to view - default to future view
+    @period = params[:period] || 'future'
     @repeating = params[:repeating] || 'on'
     @events = filter_events(@period, repeating: @repeating, site: current_site)
+    # Duration to view - default to day view if there are too many future events
+    if params[:period].to_s == '' && @events.count > 200
+      @period = 'day'
+      @events = filter_events(@period, repeating: @repeating, site: current_site)
+    end
```

Comment:
This is a little picky buuuut this kind of variable reassignment makes it much harder to understand what is going on and also you don't need to reassign events as it's identical - any chance you could refactor this to something more like:

```
should_default_to_day = params[:period].to_s == '' && @events.count > 200
@repeating = params[:repeating] || 'on'
@events = filter_events(@period, repeating: @repeating, site: current_site)
@period = should_default_to_day ? 'day' :  (params[:period] || 'future')
```

Reply:
Also also - maybe we should default to weekly here rather than daily? Lots of places have repeating events and just one on lots of days which still makes it look a bit crap by default - for example torbay

Reply:
I'll default to week, that makes it more likely to not give a blank, thanks.

> you don't need to reassign events as it's identical

I don't think that's the case because the `@events` inside the `if` block will shift the period from `future` to `day` so should *hopefully* produce a different result.

I'm afraid I'm struggling to get your snippet to produce the same result

In your code block I can't generate the `should...` var without first declaring `@events`, then again I need to set the `@period` before assigning `@events` or they won't be parametrized by it.

Reply:
ah yes! Makes sense.